core/src/main/scala/org/apache/spark/sql/execution/columnar/ExternalStoreUtils.scala

```diff
@@ -334,7 +334,12 @@ object ExternalStoreUtils {
       hikariCP: Boolean): ConnectionProperties = {
     val (user, password) = getCredentials(session)
 
-    if (!user.isEmpty && !password.isEmpty) {
+    val isSnappy = dialect match {
+      case GemFireXDDialect | GemFireXDClientDialect => true
+      case _ => false
+    }
+
+    if (!user.isEmpty && !password.isEmpty && isSnappy) {
```

Comment:
Minor: May be evaluate isSnappy only if credentials are available? Or vice versa.